### ISSUE TITLE
toBeRejected should return promise

### ISSUE THREAD

**Issue by Kampfmoehre** [earl-js/earl#64]
Hi,

[in the docs](https://earljs.dev/docs/guides/async-functions) it is described that a promise rejection can be tested with an await like this:
```js
await expect(longTermTask()).toBeRejected('Unexpected error')
```

However Eslint is complaining about an await of a non-promise ([@typescript-eslint/await-thenable](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md)). When inspecting I can see that toBeRejected returns void. Either the example is wrong and there should be not await or the toBeRejected function should return a promise.

Note: Both methods are working in my case, removing the await or ignoring the linter rule.

**Comment by krzkaczor** [comment]
It definitely needs `await` (it returns a promise) as you can see in docs https://github.com/earl-js/earl/blob/master/test/validators/toBeRejected.test.ts#L8

Type definition has a mistake - thanks for pointing that out. [This](https://github.com/earl-js/earl/blob/master/src/Expectation.ts#L58) should be `Promise<void>`. I ll fix it today 👍

**Comment by krzkaczor** [comment]
Released as `earljs@0.1.2`.

Btw. it's kinda strange that `tsc` didn't consider this a mistake as there was a mismatch in the overloaded signatures.
